frontend/src/pages/Landing/index.js

```diff
@@ -104,10 +105,28 @@ function renderReports(reports, history) {
         onClick: () => { history.push(linkTarget); },
       },
     ];
-    const contextMenuLabel = `View activity report ${displayId}`;
+
+    if (!legacyId) {
+      const downloadMenuItem = {
+        label: 'Download',
+        onClick: () => {
+          const downloadURL = getReportsDownloadURL([id]);
+          window.location.assign(downloadURL);
```

Comment:
We want to have the browser download the file, so we use `window.location.assign`. This should work because the URL in question should be set to download server-side.